### ISSUE TITLE
A cumulative time limit is not enforced

### ISSUE THREAD

**Issue by viroulep** [thewca/wca-live#5]
See below:
![cumul](https://user-images.githubusercontent.com/1007485/65134790-2ad7fb00-da05-11e9-85cb-66eeefb8fd64.png)

**Comment by jonatanklosko** [comment]
@jfly and I have discussed this, and cumulative time limits get quite tricky because it requires knowing how long someone spent on a DNF (as usually not all attempts are complete). But definitely something to think about.

**Comment by KitClement** [comment]
Maybe an optional way to enter in DNF times for rounds with cumulative limits would help check this?

**Comment by danieljames-dj** [comment]
I've created a PR (basically just to start exploring WCA Live) for this issue. If the entered score is exceeding the cumulative time limit, it will DNF the last attempts. Feel free to reject it if the solution is not right or if there is a better solution. Earlier, the delegate will come to know about this mistake only while entering the score, but here it will be little earlier.
https://github.com/thewca/wca-live/pull/51

**Comment by saranshgrover** [comment]
I think as @KitClement mentioned, it'd be helpful if we can enter times for DNF.

One solution for rounds with cumulative cutoffs is to not accept DNFs without a time, but that may be a bit too harsh, though I'd argue essential. Another problem would be keeping track of cumulative time limits across events.

**Comment by jonatanklosko** [comment]
When it comes to storing DNF times, I kinda feel that storing them in WCA Live only doesn't bring enough value compared to the need of entering them. So if we were to implement that, I'd rather see that adopted on the WCA website first (if we decide this information is valuable enough to store). If we go in this direction, then we could have scorecards better suited for that as well.

Cross-round cumulative time limits is indeed another point, but they are relatively rare and the solution suggested by @danieljames-dj is an improvement, so sounds good to me.

**Comment by c-goodyear** [comment]
A lot of competitions are staying with fixed seating, and cumulative time limits are being used for more events than they used to because it doesn't affect how many stations are available like it would with a non-fixed seating system.

It would be great to enter the time for DNFs during cumulative rounds and have it calculated only within Live.
- the Delegate looking into cumulative time limit warnings from the results submission check a lot easier, especially if the Delegate is not doing the data entry.
- rounds with cumulatives would be more viable for larger competitions
- last solves close to hitting the cumulative after a DNF in the average would be able to just be written down and not manually calculated by the judge / Delegate which takes time (and potentially a station if the judge/competitors are silly :-p)

As far as the idea to have DNF times starting to be implemented on the website as you wrote about @jonatanklosko , I think that is a bad idea because then it would encourage people to start getting "fast DNFs" on purpose, and such data serves no purpose.

I don't think you even need to display the DNF time in Live, just have them available from the Admin side. That way the average competitor doesn't even see a difference!

**Comment by dunkOnIT** [comment]
I agree with Callum that times for DNF's should be stored optionally and only for administrative purposes in WCA Live. I'll create an issue for that, with this one being dependent on it.
